### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by March1205**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,78 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
```
Comment (vsmutok) [inline app/main.py:2]:
```suggestion
    def __init__(self, km: float | int) -> None:
```

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+        self.km: float = km
+
+    def __str__(self) -> str:
+        return "Distance: {} kilometers.".format(self.km)
+
+    def __repr__(self) -> str:
+        return "Distance(km={})".format(self.km)
+
+    def __add__(self, other: "Distance") -> "Distance":
```
Comment (vsmutok) [inline app/main.py:11]:
```suggestion
    def __add__(self, other: Distance | int | float) -> Distance:
```

Comment (vsmutok) [reply app/main.py:11]: fix everywhere

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
```
Comment (vsmutok) [inline app/main.py:16]: else redundant here

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Comment by March1205** [comment]
I couldn't use '|' operator so used 'Union' instead, as my code kept failing without it. I also tried to do Distance without quotes but I kept getting NameError as name couldn't be defined and even by defining it at the end I still had to use quotes in most parts. If you could suggest something to complete this code as you told me, please do, because I might be missing some things.

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
+            return self.km == other
+        else:
+            raise TypeError("Unsupported operand type for ==")
+
+    def __le__(self, other: Union["Distance", int, float]) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        elif isinstance(other, (int, float)):
+            return self.km <= other
+        else:
```
Comment (vsmutok) [inline app/main.py:71]: `else` redundant in this case

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Comment by vsmutok** [comment]
> I couldn't use '|' operator so used 'Union' instead, as my code kept failing without it. I also tried to do Distance without quotes but I kept getting NameError as name couldn't be defined and even by defining it at the end I still had to use quotes in most parts. If you could suggest something to complete this code as you told me, please do, because I might be missing some things.


try use:
```python
from __future__ import annotations
```

**Pull request closed without merging**